frame/proxy/src/lib.rs

```diff
@@ -228,7 +228,9 @@ pub mod pallet {
 			delay: T::BlockNumber,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			Self::add_proxy_delegate(&who, delegate, proxy_type, delay)
+			Self::add_proxy_delegate(&who, delegate.clone(), proxy_type.clone(), delay)?;
```

Comment:
better to emit event within `add_proxy_delegate` so we have still have events when new proxy is added via other pallets